kite-data/kite-data-mapreduce/src/main/java/org/kitesdk/data/mapreduce/DatasetKeyOutputFormat.java

```diff
@@ -334,11 +335,20 @@ public void close(TaskAttemptContext taskAttemptContext) {
     }
   }
 
-  static class NullOutputCommitter extends OutputCommitter {
+  static class NullOutputCommitter<E> extends OutputCommitter {
     @Override
     public void setupJob(JobContext jobContext) { }
 
     @Override
+    public void commitJob(JobContext jobContext) throws IOException {
+      View<E> targetView = load(jobContext);
+
+      if(targetView instanceof ReadiableView) {
+        ((ReadiableView)targetView).signalReady();
```

Comment:
This was discussed somewhat on the [jira](https://issues.cloudera.org/browse/CDK-451?focusedCommentId=22245&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-22245), the conclusion at the time was to settle on always signaling, without a flag or configuration option.

This ties somewhat into the comment from the SignalManager around the names of the ready files. But my thinking here is that we could compact past ready signals down into some sort of a bulk storage. Some of the future thoughts around covered-by/contained-by views would require loading all the ready views, doing that in bulk should be more efficient. 

In the short term the ready signal files from MapReduce are analogous enough to SUCCESS files that I wouldn't think most would be worried about them accumulating.